samples/nrf9160/modem_shell/src/cloud/cloud_shell.c

```diff
@@ -101,7 +167,12 @@ static void nrf_cloud_event_handler(const struct nrf_cloud_evt *evt)
 	case NRF_CLOUD_EVT_RX_DATA:
 		mosh_print("NRF_CLOUD_EVT_RX_DATA");
 		if (((char *)evt->data.ptr)[0] == '{') {
-			/* Not A-GPS data */
+			/* Not A-GPS data. Check if it's a MoSh command sent from the cloud */
```

Comment:
nrf cloud now sends A-GPS error messages, which are JSON 😅 
so the check `if (((char *)evt->data.ptr)[0] == '{') {` along with the `break` might be too strict.
it would be better to move the `break` into the `cmd_found` block.

Reply:
Or maybe directly try to parse as json mosh cmd? I.e. it should fail, if data is a-gps data or a-gps error json and then it will go a-gps data handling.

Reply:
Or maybe directly try to parse as json mosh cmd? I.e. it should fail, if data is a-gps data or a-gps error json and then it will go a-gps data handling.

Reply:
I moved break into the cmd_found block. I will keep if (((char *)evt->data.ptr)[0] == '{') though and not try to parse everything as json to avoid showing a parsing error (from cloud_shell_parse_mosh_cmd()) in case of A-GPS data reception.